### PR TITLE
Blog post on Panther AWS monitoring

### PR DESCRIPTION
**Pull request by glerb**

### CODE REVIEW

File: _posts/2020-04-28-monitoring-aws-with-panther.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+The monitoring challenges for such a scenario are significant. There aren't necessarily "more" monitoring datapoints, but the speed with which infra can be created tends to result in infrastructure getting way out over its skis with respect to monitoring. Furthermore, since many barriers to entry for doing stupid things have been lowered to the point of non-existence, monitoring is the last great hope of maintaining control over a cloud environment. While access controls can still provide some guardrails, the flexibility that all engineers need to do their jobs requires that they have the ability to do "dangerous" things that they've never had to do before. The true definition of "full stack" has expanded.
```
[the remark this answers is not in the document]
Comment (bartdegoede) [reply _posts/2020-04-28-monitoring-aws-with-panther.md:13]: `infra` => `infrastructure`

File: _posts/2020-04-28-monitoring-aws-with-panther.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+It was also very difficult to precisely define user permissions for infrastructure work in legacy environments. At Scribd, this resulted in a small cadre of sysadmins having root access to everything and no one else having access to anything. With AWS IAM, the Instance Metadata Service (IMDS), or some combination of the two, access permissions can be easily set in a consistent format for any type of infrastructure resource.
+
+# A new solution to an old problem
+
+Unfortunately, native AWS services can't fully take advantage of the power that its event logging and permissions settings provide. Scribd needed a monitoring solution that could keep up with our expanding infra, alerting us when certain events occurred or when permissions were set inappropriately.
```
[the remark this answers is not in the document]
Comment (bartdegoede) [reply _posts/2020-04-28-monitoring-aws-with-panther.md:25]: `that alerts us when certain events occur or when permissions are set inappropriately`

File: _posts/2020-04-28-monitoring-aws-with-panther.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+We recently deployed the [Panther](https://www.runpanther.io) monitoring system in several AWS accounts. Right out of the box, it enables us to see certain near-real-time changes in these accounts, such as changes in security groups, using AWS EventBridge as a base. It also performs a daily configuration check for a defined set of configuration options, such as S3 buckets' public writeability and the existence of MFA on root IAM user accounts. We currently have alerts for events and policy failures sent to a dedicated Slack channel. There is also a historical CloudTrail search functionality that makes hunting for events easy. The newest feature allows pivoting across multiple log sources. In other words, "correlations". That's what a SIEM is built for.
```
Comment (bartdegoede) [inline _posts/2020-04-28-monitoring-aws-with-panther.md:27]: Link `SIEM` to https://en.wikipedia.org/wiki/Security_information_and_event_management :-)

**Comment by rtyler** [comment]
I'll update the date and prepare for publication after some more copy editing and fixing has been done.

File: _posts/2020-04-28-monitoring-aws-with-panther.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+# We're moving!
+
+Scribd is in the midst of migrating our entire infrastructure from a legacy data center to AWS. Things are moving fast. We've given developer teams nearly complete access to their own AWS accounts. We've labeled these accounts "development environments" and haven't created any cross-connections between them and production system accounts, but developers still have a lot of power, much more than they had in the legacy environment.
+
+The AWS cloud has a few important saving graces to manage the new chaos, for which there isn't really an analogue in traditional data centers: universal event logging in a standard format for all resources, and highly granular permissions settings in a consistent format. Universal event logging in legacy centers was usually an asymptote that mortal sysadmins and security engineers could never actually reach. This was due to the inability to get complete data off of a device, inability to properly parse the data that could be exported, or a combination of both. AWS CloudTrail solves both problems.
```
[the remark this answers is not in the document]
Comment (glerb) [reply _posts/2020-04-28-monitoring-aws-with-panther.md:19]: https://english.stackexchange.com/questions/162281/is-a-comma-necessary-before-for-which

"Chaos" is "all chaos", not just a subset.

File: _posts/2020-04-28-monitoring-aws-with-panther.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (glerb) [reply _posts/2020-04-28-monitoring-aws-with-panther.md:25]: > `needed` => `needs`

but we have one now.



> `that alerts us when certain events occur or when permissions are set inappropriately`

I think either one is ok.

[the remark this answers is not in the document]
Comment (bartdegoede) [reply _posts/2020-04-28-monitoring-aws-with-panther.md:25]: Just tryin' to keep the tense appropriate ;-)

File: _posts/2020-04-28-monitoring-aws-with-panther.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+layout: post
+title: "Using Panther to monitor AWS infrastructure"
+authors:
+- paha
+tags:
+- monitoring
+team: Security Engineering
+---
+
+Before widespread cloud usage, it was uncommon for one person to be present for the entire datacenter development lifecycle. Very few people knew how to design and build a datacenter from scratch while ensuring appropriate security configuration settings were set, on top of rigging up monitoring. It was even more uncommon for non-sysadmins to have any involvement in data center infrastructure construction or ongoing refinement. The cloud is very different. It only takes seconds to create an entire infrastructure from a template. And even devs are doing it!
```
[the remark this answers is not in the document]
Comment (glerb) [reply _posts/2020-04-28-monitoring-aws-with-panther.md]: Done.

File: _posts/2020-04-28-monitoring-aws-with-panther.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (glerb) [reply _posts/2020-04-28-monitoring-aws-with-panther.md]: Done.

File: _posts/2020-04-28-monitoring-aws-with-panther.md (hunk 3 above)
Comment (glerb) [reply _posts/2020-04-28-monitoring-aws-with-panther.md]: Done.

**Comment by glerb** [comment]
Not sure why Security Engineering isn't showing up in the bottom left pane. Source code for those list items looks generated.

**Comment by rtyler** [comment]
@glerb https://github.com/scribd/scribd.github.io/tree/master/_category and https://github.com/scribd/scribd.github.io/blob/master/_data/teams.yml need some data. They are generated, but not automatically off the front-matter in the blog posts unfortunately

**Comment by rtyler** [comment]
merged the old fashioned way

**Pull request closed without merging**